### PR TITLE
feat(timeseries): support advanced filters, partitions [DEVX-162]

### PR DESCRIPTION
**Pull request by polomani**

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -1660,19 +1664,33 @@ export interface TimeSeriesUpdateByExternalId
...
    */
   rootAssetIds?: CogniteInternalId[];
+  externalIdPrefix?: ExternalIdPrefix;
+}
+
+export interface TimeseriesFilterQuery extends FilterQuery {
+  filter?: TimeseriesFilterProps;
+}
+
+export interface TimeseriesFilter extends TimeseriesFilterProps, Limit {
```
Comment (BugGambit) [inline src/types/types.ts]: Missing cursor

**Review by BugGambit — APPROVED** [review]
LGTM

File: src/types/types.ts [hunk 2]
```diff
@@ -313,15 +313,13 @@ export interface AssetListScope extends AssetFilter, FilterQuery {
...
+export interface SequenceFilter extends CreatedAndLastUpdatedTimeFilter {
   filter?: {
     name?: SequenceName;
-    externalIdPrefix?: CogniteExternalId;
+    externalIdPrefix?: ExternalIdPrefix;
     metadata?: Metadata;
     assetIds?: CogniteInternalId[];
     rootAssetIds?: IdEither[];
-    createdTime?: DateRange;
-    lastUpdatedTime?: DateRange;
```
Comment (polomani) [inline src/types/types.ts]: I just noticed that I messed up here, don't merge

File: src/types/types.ts [hunk 3]
```diff
@@ -784,10 +782,8 @@ export interface FileFilter extends Limit {
      */
     assetIds?: CogniteInternalId[];
     source?: string;
-    createdTime?: DateRange;
-    lastUpdatedTime?: DateRange;
```
Comment (polomani) [inline src/types/types.ts:788]: and here as well

**Pull request merged**